src/shims/thread.rs

```diff
@@ -85,44 +83,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
 
         let thread_id = this.get_active_thread()?;
-        this.write_scalar(Scalar::from_uint(thread_id.index() as u128, dest.layout.size), dest)
+        this.write_scalar(Scalar::from_uint(thread_id.to_u128(), dest.layout.size), dest)
     }
 
     fn prctl(
         &mut self,
         option: OpTy<'tcx, Tag>,
         arg2: OpTy<'tcx, Tag>,
-        arg3: OpTy<'tcx, Tag>,
-        arg4: OpTy<'tcx, Tag>,
-        arg5: OpTy<'tcx, Tag>,
+        _arg3: OpTy<'tcx, Tag>,
+        _arg4: OpTy<'tcx, Tag>,
+        _arg5: OpTy<'tcx, Tag>,
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
-        // prctl last 5 arguments are declared as variadic. Therefore, we need
-        // to check their types manually.
-        let c_long_size = this.libc_ty_layout("c_long")?.size.bytes();
-        let check_arg = |arg: OpTy<'tcx, Tag>| -> InterpResult<'tcx> {
-            match this.read_scalar(arg)?.not_undef()? {
-                Scalar::Raw { size, .. } if u64::from(size) == c_long_size => Ok(()),
-                _ => throw_ub_format!("an argument of unsupported type was passed to prctl"),
-            }
-        };
-        check_arg(arg2)?;
-        check_arg(arg3)?;
-        check_arg(arg4)?;
-        check_arg(arg5)?;
-
-        let option = this.read_scalar(option)?.not_undef()?.to_i32()?;
+        let option = this.read_scalar(option)?.to_i32()?;
         if option == this.eval_libc_i32("PR_SET_NAME")? {
             let address = this.read_scalar(arg2)?.not_undef()?;
             let name = this.memory.read_c_str(address)?.to_owned();
             this.set_active_thread_name(name)?;
         } else if option == this.eval_libc_i32("PR_GET_NAME")? {
             let address = this.read_scalar(arg2)?.not_undef()?;
-            let name = this.get_active_thread_name()?;
+            let name = this.get_active_thread_name()?.to_vec();
             this.memory.write_bytes(address, name)?;
```

Comment:
Hm, we could avoid the allocation entirely here if we didn't call `this.get_active_thread_name` but some method on `this.machine` instead... not sure if that's worth it though.

Reply:
Calling something on `this.machine` would make code inconsistent and so I am not sure it is a good idea to change this.